pkg/ctl/create/cluster.go

```diff
@@ -360,14 +360,17 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 				return fmt.Errorf("failed to create addons")
 			}
 		}
-		config := kubeconfig.NewForKubectl(cfg, ctl.GetUsername(), params.AuthenticatorRoleARN, ctl.Provider.Profile())
-		kubeConfigBytes, err := runtime.Encode(clientcmdlatest.Codec, config)
-		if err != nil {
-			return errors.Wrap(err, "generating kubeconfig")
-		}
+
 		// After we have the cluster config and all the nodes are done, we install Karpenter if necessary.
-		if err := installKarpenter(ctl, cfg, stackManager, clientSet, kubernetes.NewRESTClientGetter("karpenter", string(kubeConfigBytes))); err != nil {
-			return err
+		if cfg.Karpenter != nil {
```

Comment:
Just a small refactor so we don't create this config thing if Karpenter is not required at all